code/game/machinery/turret_control.dm

```diff
@@ -180,9 +180,32 @@
 	TC.ailock = ailock
 
 	if(istype(control_area))
+		var/lethal_only = 0 // To check if there is a lethal turret in the area.
+		var/both_mode = 0
 		for (var/obj/machinery/porta_turret/aTurret in control_area)
+			// If turret only has lethal mode - lock switching modes
+			if(aTurret.egun == 0)
+				lethal_only = 1
+				egun = 0
+			else
+				both_mode = 1
```

Comment:
This is susceptible to race conditions. Suppose before the last turret you have stun only turrets. But then the last turret is capable of both. Suddenly, both states are valid.

Better way:
```DM
egun = TRUE
for (var/obj/machinery/porta_turret/T in control_area)
	if (!T.egun)
		egun = FALSE
		break // Because we really don't care after this point.
```

Reply:
@skull132 That is not True, we care about both conditions, not one. If you look at the code below this if statement you will see. So it is looking for specific case when we have mix of one mode and two modes turrets together, and also it by this logic accounts if we only have lethal. So in your example if we before last turret we have stun only(and it doesn't check for stun only, just laser only and both mode). So yeah, need to make it account for mix of both and stun only turrets.